services/ui-src/src/routes/ReviewSubmit/ReviewSubmit.tsx

```diff
@@ -1,29 +1,215 @@
+import { useContext, useEffect } from "react";
 // components
-import { Box, Heading, Text } from "@chakra-ui/react";
-import { ReportPage } from "components";
-import verbiage from "verbiage/pages/mcpar/mcpar-review-and-submit";
+import {
+  Box,
+  Button,
+  Flex,
+  Image,
+  Heading,
+  Text,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { Icon, Modal, ReportContext, ReportPage, Sidebar } from "components";
+// types
+import { ReportStatus, UserRoles } from "types";
+// utils
+import { useUser } from "utils";
+// form data
+import reviewVerbiage from "verbiage/pages/mcpar/mcpar-review-and-submit";
+import successVerbiage from "verbiage/pages/mcpar/mcpar-successful-submit";
+// assets
+import checkIcon from "assets/icons/icon_check_circle.png";
 
 export const ReviewSubmit = () => {
-  const { intro } = verbiage;
+  const { reportStatus, fetchReportStatus, updateReportStatus } =
+    useContext(ReportContext);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  // get user's state
+  const { user } = useUser();
+  const { state, userRole } = user ?? {};
+  const reportYear = "2022";
+  const reportKey = `${state}${reportYear}`;
+
+  // TODO: get real program name per report
+  const programName = "tempName";
+
+  const reportDetails = {
+    key: reportKey,
+    programName: programName,
+  };
+
+  useEffect(() => {
+    fetchReportStatus(reportDetails);
+  }, []);
+
+  const submitForm = () => {
+    // TODO: Add check to make sure user filled out the form
+    if (userRole === UserRoles.STATE_USER || userRole === UserRoles.STATE_REP) {
+      updateReportStatus(reportDetails, ReportStatus.COMPLETED);
+    }
+    // TODO: Handle Error for other users coming on to the page
+    onClose();
+  };
+
   return (
     <ReportPage data-testid="review-and-submit-view">
+      <Flex sx={sx.pageContainer}>
+        <Sidebar />
+        {reportStatus?.status == ReportStatus.COMPLETED ? (
+          <SuccessMessage
+            programName={programName}
+            date={reportStatus?.lastAltered}
+            givenName={user?.given_name}
+            familyName={user?.family_name}
+          />
+        ) : (
+          <ReadyToSubmit
+            submitForm={submitForm}
+            isOpen={isOpen}
+            onOpen={onOpen}
+            onClose={onClose}
+          />
+        )}
+      </Flex>
+    </ReportPage>
+  );
+};
+
+const ReadyToSubmit = ({
+  submitForm,
+  isOpen,
+  onOpen,
+  onClose,
+}: ReadyToSubmitProps) => {
+  const { intro, modal, pageLink } = reviewVerbiage;
+
+  return (
+    <Flex sx={sx.contentContainer} data-testid="ready-view">
```

Comment:
Nit: Wherever possible, it's best practice to only place `data-testid` in the test files, where it is usually spread onto the element via props, instead of placing the ids in the component code directly.